packages/server/src/fhir/references.ts

```diff
@@ -1,105 +1,128 @@
 import {
+  badRequest,
+  crawlResource,
+  createReference,
   createStructureIssue,
-  getTypedPropertyValue,
   normalizeErrorString,
   OperationOutcomeError,
+  parseSearchRequest,
   PropertyType,
-  toTypedValue,
   TypedValue,
 } from '@medplum/core';
-import { OperationOutcomeIssue, Project, Reference, Resource } from '@medplum/fhirtypes';
+import { OperationOutcomeIssue, Reference, Resource } from '@medplum/fhirtypes';
 import { randomUUID } from 'crypto';
-import { getSystemRepo } from './repo';
+import { Repository } from './repo';
 
-export async function validateReferences<T extends Resource>(
-  resource: T,
-  allowedProjects?: Project['id'][]
+async function validateReference(
+  reference: Reference,
+  repo: Repository,
```

Comment:
Another minor comment - I was always told to make required "context" args first in the args list.  In general, that has meant putting `repo` early.  Apologies for bike sheddy comment.  I guess I'd be more curious if the team has opinions on this, and whether we could get alignment.  It always feels nice when arg order feels natural, but that's obviously a super subjective thing.

This is an internal api, so totally non-blocking.